After v2.4.0 upgrade getting occasional nil pointer panics

Hi - Last week I upgraded to latest gubernator version v2.4.0 from the old repo (https://github.com/mailgun/gubernator) and since then some servers have occasionally hit a nil pointer (panic). Previously we were running v2.2.1, never saw this issue.

Starting investigation, adding here in case core team can spot an obvious problem / cause, stack below. It looks like it is possible to hit gubernator.go:334 with either resp or err being nil?

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1d4bc8e] github.com/mailgun/gubernator/v2.(*V1Instance).asyncRequest(0xc00034cf00, {0x3402690?, 0xc00a770510?}, 0xc0080d8f40)
github.com/mailgun/gubernator/v2.(*V1Instance).asyncRequest(0xc00034cf00, {0x3402690?, 0xc00a770510?}, 0xc0080d8f40)
[redacted-go-project-name]/vendor/github.com/mailgun/gubernator/v2/gubernator.go:334 +0x104e
github.com/mailgun/gubernator/v2.(*V1Instance).GetRateLimits in goroutine 49872128
[redacted-go-project-name]/vendor/github.com/mailgun/gubernator/v2/gubernator.go:275 +0xa25
Main process exited, code=exited, status=2/INVALIDARGUMENT
```

--- a/functional_test.go
+++ b/functional_test.go
@@ -743,11 +743,12 @@ func TestLeakyBucketGregorian(t *testing.T) {
 		},
 	}
 
-	// Truncate to the nearest minute
+	// Truncate to the nearest minute.
 	now := clock.Now()
-	now = now.Truncate(1 * time.Minute)
-	// So we don't start on the minute boundary
-	now = now.Add(time.Millisecond * 100)
+	trunc := now.Truncate(time.Hour)
+	trunc = now.Add(now.Sub(trunc))
+	clock.Advance(now.Sub(trunc))
+
 	name := t.Name()
 	key := guber.RandomString(10)
 

--- a/gubernator.go
+++ b/gubernator.go
@@ -330,7 +330,7 @@ func (s *V1Instance) asyncRequest(ctx context.Context, req *AsyncReq) {
 				WithField("key", req.Key).
 				Error("GetPeer() returned peer that is not connected")
 			countError(err, "Peer not connected")
-			err = errors.Wrapf(err, "GetPeer() keeps returning peers that are not connected for '%s'", req.Key)
+			err = fmt.Errorf("GetPeer() keeps returning peers that are not connected for '%s': %w", req.Key, err)
 			resp.Resp = &RateLimitResp{Error: err.Error()}
 			break
 		}
@@ -344,7 +344,7 @@ func (s *V1Instance) asyncRequest(ctx context.Context, req *AsyncReq) {
 						WithError(err).
 						WithField("key", req.Key).
 						Error("Error applying rate limit")
-					err = errors.Wrapf(err, "Error in getLocalRateLimit for '%s'", req.Key)
+					err = fmt.Errorf("during getLocalRateLimit() for '%s': %w", req.Key, err)
 					resp.Resp = &RateLimitResp{Error: err.Error()}
 				}
 				break
@@ -352,17 +352,18 @@ func (s *V1Instance) asyncRequest(ctx context.Context, req *AsyncReq) {
 		}
 
 		// Make an RPC call to the peer that owns this rate limit
-		r, err := req.Peer.GetPeerRateLimit(ctx, req.Req)
+		var r *RateLimitResp
+		r, err = req.Peer.GetPeerRateLimit(ctx, req.Req)
 		if err != nil {
 			if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
 				attempts++
 				metricBatchSendRetries.WithLabelValues(req.Req.Name).Inc()
 				req.Peer, err = s.GetPeer(ctx, req.Key)
 				if err != nil {
-					errPart := fmt.Sprintf("Error finding peer that owns rate limit '%s'", req.Key)
+					errPart := fmt.Sprintf("while finding peer that owns rate limit '%s'", req.Key)
 					s.log.WithContext(ctx).WithError(err).WithField("key", req.Key).Error(errPart)
-					countError(err, "Error in GetPeer")
-					err = errors.Wrap(err, errPart)
+					countError(err, "during GetPeer()")
+					err = fmt.Errorf("%s: %w", errPart, err)
 					resp.Resp = &RateLimitResp{Error: err.Error()}
 					break
 				}
@@ -371,7 +372,7 @@ func (s *V1Instance) asyncRequest(ctx context.Context, req *AsyncReq) {
 
 			// Not calling `countError()` because we expect the remote end to
 			// report this error.
-			err = errors.Wrap(err, fmt.Sprintf("Error while fetching rate limit '%s' from peer", req.Key))
+			err = fmt.Errorf("while fetching rate limit '%s' from peer: %w", req.Key, err)
 			resp.Resp = &RateLimitResp{Error: err.Error()}
 			break
 		}

